src/main/java/it/pagopa/pn/delivery/svc/search/NotificationRetrieverService.java

```diff
@@ -264,6 +264,9 @@ public InternalNotification getNotificationInformation(String iun, boolean withT
 
 		if (optNotification.isPresent()) {
 			InternalNotification notification = optNotification.get();
+			if ( requestBySender && StringUtils.hasText( senderId ) ) {
```

Comment:
Perché metti in and anche StringUtils.hasText( senderId ) ?
Significa che se non c'è il senderId non viene eseguito il test, mentre se non c'è il test deve fallire.